RELEASE.md

```diff
@@ -0,0 +1,54 @@
+release type: minor
+
+Permissions classes now use a `FieldExtension`. The new preferred way to add permissions
+is to use the `PermissionsExtension` class:
+
+```python
+import strawberry
+from strawberry.permission import PermissionExtension, BasePermission
+
+
+class IsAuthorized(BasePermission):
+    message = "User is not authorized"
+    error_extensions = {"code": "UNAUTHORIZED"}
+
+    def has_permission(self, source, info, **kwargs) -> bool:
```

Comment:
ah, the other day I was thinking about refactoring this to remove the need of passing source, info and kwargs.

What do you think if we move them to be properties on the class? I think that would make the code much nicer 😊

Reply:
should we also allow users to set a status code somehow?

related: https://github.com/strawberry-graphql/strawberry/issues/2399

Reply:
@patrick91 since source and info change on every request, but the Permission class persists over requests I am not sure if this is a good idea. Take an async permission which yields to another coroutine. Then suddenly the source and info class properties change. We can make them optional for the user if you prefer, but `info.context` is often very crucial for the permission checks.

Reply:
we could have the same approach we have for resolvers for root/info, but not sure how we could do it for kwargs 🤔 

Also, I really don't like the fact we need to add kwargs there :D 

Oh I was also thinking about being able to check permission after resolving, would that be a valid use case? /cc @bellini666 

maybe after I'm done travelling we can pair on this?

Reply:
I agree on both points, but not sure it's in scope for this PR. Everything here is backwards-compatible. If we change the structure of `has_permission` it might break things for other users.
Already discussed permission after resolving with @bellini666, that's also something separate we can tackle building up on this.

Let's get this started after your travels 😊

Reply:
> I agree on both points, but not sure it's in scope for this PR. Everything here is backwards-compatible. If we change the structure of has_permission it might break things for other users.

makes sense, I didn't check all the code just yet I thought we were going to replace how users would use permissions 😊